tooling/nargo_toml/src/lib.rs

```diff
@@ -518,7 +519,10 @@ pub fn resolve_workspace_from_toml(
     current_compiler_version: Option<String>,
 ) -> Result<Workspace, ManifestError> {
     let nargo_toml = read_toml(toml_path)?;
+    let lock = lock_git_deps().expect("Failed to lock git dependencies cache");
     let workspace = toml_to_workspace(nargo_toml, package_selection)?;
+    lock.unlock().expect("Failed to unlock git dependencies cache");
```

Comment:
I reckon it would be prudent to unlock the file before returning an error if `toml_to_workspace` fails.

Reply:
Ah good point. I'm being lazy here. Will fix